src/main/scala/firrtl/passes/CheckWidths.scala

```diff
@@ -109,46 +109,65 @@ object CheckWidths extends Pass with PreservesAll[Transform] {
-            case other =>
+    def check_width_e_inner(info: Info, target: Target, expr: Expression): Unit = {
+      expr match {
+        case e @ UIntLiteral(v, w: IntWidth) if math.max(1, v.bitLength) > w.width =>
+          errors.append(new WidthTooSmall(info, target.serialize, v))
+        case e @ SIntLiteral(v, w: IntWidth) if v.bitLength + 1 > w.width =>
+          errors.append(new WidthTooSmall(info, target.serialize, v))
+        case e @ DoPrim(op, Seq(a, b), _, tpe) =>
+          (op, a.tpe, b.tpe) match {
+            case (Squeeze, IntervalType(Closed(la), Closed(ua), _), IntervalType(Closed(lb), Closed(ub), _)) if (ua < lb) || (ub < la) =>
+              errors.append(new DisjointSqueeze(info, target.serialize, e))
+            case (Dshl, at, bt) if (hasWidth(at) && bitWidth(bt) >= DshlMaxWidth) =>
+              errors.append(new DshlTooBig(info, target.serialize))
+            case _ =>
           }
-        case DoPrim(Bits, Seq(a), Seq(hi, lo), _) if (hasWidth(a.tpe) && bitWidth(a.tpe) <= hi) =>
-          errors.append(new BitsWidthException(info, target.serialize, hi, bitWidth(a.tpe), e.serialize))
-        case DoPrim(Head, Seq(a), Seq(n), _) if (hasWidth(a.tpe) && bitWidth(a.tpe) < n) =>
-          errors.append(new HeadWidthException(info, target.serialize, n, bitWidth(a.tpe)))
-        case DoPrim(Tail, Seq(a), Seq(n), _) if (hasWidth(a.tpe) && bitWidth(a.tpe) < n) =>
-          errors.append(new TailWidthException(info, target.serialize, n, bitWidth(a.tpe)))
-        case DoPrim(Dshl, Seq(a, b), _, _) if (hasWidth(a.tpe) && bitWidth(b.tpe) >= DshlMaxWidth) =>
-          errors.append(new DshlTooBig(info, target.serialize))
-        case DoPrim(AsClock, Seq(a), _, _) if (bitWidth(a.tpe) != 1) =>
-          errors.append(new MultiBitAsClock(info, target.serialize))
-        case DoPrim(AsAsyncReset, Seq(a), _, _) if (bitWidth(a.tpe) != 1) =>
-          errors.append(new MultiBitAsAsyncReset(info, target.serialize))
+        case e @ DoPrim(op, Seq(a), consts, _) =>
+          (op, consts) match {
+            case (Bits, Seq(hi, lo)) if (hasWidth(a.tpe) && bitWidth(a.tpe) <= hi) =>
+              errors.append(new BitsWidthException(info, target.serialize, hi, bitWidth(a.tpe), e.serialize))
+            case (Head, Seq(n)) if (hasWidth(a.tpe) && bitWidth(a.tpe) < n) =>
+              errors.append(new HeadWidthException(info, target.serialize, n, bitWidth(a.tpe)))
+            case (Tail, Seq(n)) if (hasWidth(a.tpe) && bitWidth(a.tpe) < n) =>
+              errors.append(new TailWidthException(info, target.serialize, n, bitWidth(a.tpe)))
+            case (AsClock, _) if (bitWidth(a.tpe) != 1) =>
+              errors.append(new MultiBitAsClock(info, target.serialize))
+            case (AsAsyncReset, _) if (bitWidth(a.tpe) != 1) =>
+              errors.append(new MultiBitAsAsyncReset(info, target.serialize))
+            case _ =>
+          }
+        case _ =>
```

Comment:
Related to comment on `check_width_e`, can you include a comment for this case on why the expressions are not recursively traversed (no point as others won't be checked).